lib/spandx/js/parsers/yarn.rb

```diff
@@ -3,239 +3,34 @@
 module Spandx
   module Js
     module Parsers
-      class Token
-        attr_accessor :type, :value
-
-        def initialize(type, value)
-          @type = type
-          @value = value
-        end
-      end
-
       class Yarn < ::Spandx::Core::Parser
-        attr_accessor :tokens, :token
-
-        # lockfile version, bump whenever we make backwards incompatible changes
-        LOCKFILE_VERSION = 1
-        TOKEN_TYPES = {
-          boolean: 'BOOLEAN',
-          string: 'STRING',
-          identifier: 'IDENTIFIER',
-          eof: 'EOF',
-          colon: 'COLON',
-          newline: 'NEWLINE',
-          comment: 'COMMENT',
-          indent: 'INDENT',
-          invalid: 'INVALID',
-          number: 'NUMBER',
-          comma: 'COMMA',
-        }.freeze
+        START_OF_DEPENDENCY_REGEX = %r{^"?(?<name>(@|\w|-|\.|/)+)@}i.freeze
```

Comment:
@mokhan 

Can we also get `resolved` section from `yarn.lock` and put into dependency ?  Something like following (haven't tested)


```ruby 
yield ::Spandx::Core::Dependency.new(
                name: matches[:name].gsub(/"/, ''),
                version: io.readline.split(' ')[-1].gsub(/"/, '')
                meta: { base_registry_url:  io.readline.strip.split(' ')[-1]  }
              )

def get_registry_url(input)
    require 'uri'
    uri = URI.parse(input)
    "#{uri.scheme}://#{uri.host}"
end
```

In case another registry is used we could make an api call to that in order to fetch license information. WDYT ?

Reply:
Oh yeah, totally. I have an idea.